tox.ini

```diff
@@ -172,3 +175,9 @@ commands =
                 --hidden-import pytest_doctestplus.plugin \
                 --hidden-import pytest_mpl.plugin
     ./run_astropy_tests --astropy-root {toxinidir}
+
+[testenv:typecheck]
+changedir = {tox_root} # restore default
+dependency_groups = typecheck
+commands =
+  basedpyright --stats
```

Comment:
Added `stats` here so one gets at least an idea of how many files are actually checked...